observation/mapper.go

```diff
@@ -0,0 +1,76 @@
+package observation
+
+import (
+	"encoding/csv"
+	inputcsv "github.com/ONSdigital/dp-observation-importer/csv"
+	"strings"
+)
+
+// Mapper interprets a CSV line and returns an observation instance.
+type Mapper struct {
+	dimensionCache DimensionHeaderCache
+}
+
+// DimensionHeaderCache provides the an array of dimension names to define the order of dimensions (v4 format)
+type DimensionHeaderCache interface {
+	GetOrder(instanceID string) ([]string, error)
+}
+
+// NewMapper returns a new Mapper instance
+func NewMapper(dimensionOrderCache DimensionHeaderCache) *Mapper {
+	return &Mapper{
+		dimensionCache: dimensionOrderCache,
+	}
+}
+
+// Map the given CSV row to an observation instance.
+func (mapper *Mapper) Map(row string, instanceID string) (*Observation, error) {
+
+	headerRow, err := mapper.dimensionCache.GetOrder(instanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	header := inputcsv.NewHeader(headerRow)
+	dimensionOffset, err := header.DimensionOffset()
+	if err != nil {
+		return nil, err
+	}
+
+	var dimensions []*DimensionOption
+	csv := csv.NewReader(strings.NewReader(row))
+	csvRow, err := csv.Read()
+	if err != nil {
+		return nil, err
+	}
+
+	// if we want to support time being in any column we need to look at the header to see what column time is in.
+	// as it
+	timeDimensionOffset := dimensionOffset // Assume time is always first
```

Comment:
Added a todo to review this. I don't want to just add it if we don't need to. It could end up being a lot of work to calculate this for every row.